src/inc/corjithost.h

```diff
@@ -43,6 +43,8 @@ class ICorJitHost
     virtual void freeStringConfigValue(
         const wchar_t* value
         ) = 0;
+
+    virtual ~ICorJitHost() = 0 {}
 };
```

Comment:
This interface declaration should not have a destructor; it should only have named pure virtual methods. If you want a destructor, virtual or not, don't you want it on the SuperPMI JitHost class?

Reply:
Because JitInstance keeps ICorJitHost* jitHost we need destructor for the base class.

> This  interface declaration should not have a destructor;

I don't agree that C++ abstract class should not have destructor.

Reply:
SuperPMI JitInstance knows it is a JitHost, not an ICorJitHost.

Reply:
`src\ToolBox\superpmi\superpmi\jitinstance.h::JitInstance` keeps `ICorJitHost*   jitHost;`

We can change its type to JitHost, but there are 5 different JitHost classes, so this interface type was created to be used as declaration type for such fields. I do not want to change it.

I insist that we must declare base class constructor as virtual. 

Yes, it will change virtual method table size for ICorJitHost, but I do not see any risks here.

Reply:
I'm also of the opinion that this interface shouldn't have a destructor: the lifetime of an `ICorJitHost` instance is not bound by the JIT, which should be the only thing that actually holds an `ICorJitHost` instance.

> We can change its type to JitHost, but there are 5 different JitHost classes

IIRC there should only be a single implementation of `JitHost` per compiled binary.

Reply:
Ok, I am sure you know that, for global `ICorJitHost*   g_jitHost;` during the process end we call wrong ~ICorJitHost without ~JitHost nowadays. I understand that in this case it doesn't matter, because destructors are empty. I think it is just a bad practice without any advantages.